ledger/acctupdates.go

```diff
@@ -1045,10 +1058,22 @@ func (au *accountUpdates) commitRound(offset uint64, dbRound basics.Round, lookb
 				return
 			}
 		}
+
+		// if everytyhing went well,
+		if err == nil {
+			// take the lock, as we're going to attempt to commit the transaction to database.
+			au.accountsMu.Lock()
+			lockTaken = true
+		}
 		return err
 	})
 
 	if err != nil {
+		// if we failed during the commit, undo the lock.
+		if lockTaken {
+			au.accountsMu.Unlock()
```

Comment:
`lockTaken = false`

Reply:
why ? we're exiting from commitRound, why should we care about local variables anymore ?

Reply:
for the purpose of the check you just added, the variable should stay up to date

Reply:
but I suppose it doesn't actually matter.